db/compaction/compaction_picker_test.cc

```diff
@@ -3656,6 +3656,63 @@ TEST_P(PerKeyPlacementCompactionPickerTest, NormalCompactionOverlapUniversal) {
                 input_files, 5));
 }
 
+TEST_P(PerKeyPlacementCompactionPickerTest, PenultimateOverlapUniversal) {
+  // This test is make sure the Tiered compaction would lock whole range of
+  // both output level and penultimate level
+  if (enable_per_key_placement_) {
+    ioptions_.preclude_last_level_data_seconds = 10000;
+  }
+
+  int num_levels = ioptions_.num_levels;
+  ioptions_.compaction_style = kCompactionStyleUniversal;
+  UniversalCompactionPicker universal_compaction_picker(ioptions_, &icmp_);
+  NewVersionStorage(num_levels, kCompactionStyleUniversal);
+
+  // L4:   [200, 220]  [230, 250]       [360, 380]
+  // L5:
+  // L6: [101,                    351]
+  Add(4, 40U, "200", "220", 60000000U);
+  Add(4, 41U, "230", "250", 60000000U);
+  Add(4, 42U, "360", "380", 60000000U);
+  Add(6, 50U, "101", "351", 60000000U);
+  UpdateVersionStorageInfo();
```

Comment:
I think it is important to cover the scenario where there are only files in the last level, but the penultimate level is still locked and further compaction trying to put files there will not be scheduled. 

For example, there are only files in the last level, and a periodic compaction is triggered to recompact this level. This compaction runs for very long that many L0 compactions are created during the time and a compaction is to be scheduled to put files in L(last-1). This should not be allowed to execute.

Reply:
Only last level compaction doesn't support move to penultimate level yet, will create a new PR specifically for that.